src/Features/Blockcore.Features.Wallet/Types/Wallet.cs

```diff
@@ -27,17 +28,17 @@ public class Wallet
         public const int SpecialPurposeAccountIndexesStart = 100_000_000;
 
         /// <summary>Filter for identifying normal wallet accounts.</summary>
-        public static Func<HdAccount, bool> NormalAccounts = a => a.Index < SpecialPurposeAccountIndexesStart;
+        public static Func<IHdAccount, bool> NormalAccounts = a => a.Index < SpecialPurposeAccountIndexesStart;
 
         /// <summary>Filter for all wallet accounts.</summary>
-        public static Func<HdAccount, bool> AllAccounts = a => true;
+        public static Func<IHdAccount, bool> AllAccounts = a => true;
```

Comment:
Honestly this PR would be much easier to review if the interfaces over account and address was not created, is that really necessary?